### ISSUE TITLE
Initial work

### ISSUE THREAD

**Issue by jeau** [jeau/tiddlywiki-sandstorm#1]
@koivunej  

Few days ago, to learn Sandstorm and rediscover TiddlyWiki, I began a basic implementation of TiddlyWiki5 as a Sandstorm app. On TiddlyWiki GoogleGroup, @Jermolene told me that you have already started this work. It was discussed in this thread on GitHub : 
https://github.com/Jermolene/TiddlyWiki5/issues/2487 

Have you made any progress toward packaging TW5 ? We could compare and merge our codes.

**Comment by koivunej** [comment]
Yeah I might have found something more interesting while waiting for my sandstorm patch to get to stable :) I have since forgotten to periodically check for this **but** if your version now works on latest `vagrant-spk` and stable sandstorm it'd seem like my patch is in the stable already.

> We could compare and merge our codes.

Looking at your `setup.sh` which installs npm and tiddlywiki it's exactly the same solution as I had. You've gone ways beyond mine solution with app-graphics :)

The more general issues I was thinking about after I managed to get my simple PR into sandstorm was (using blockquotes for the "issues"):

> If I publish the sandstorm/tiddlywiki app into sandstorm marketplace, will I be locked in as a maintainer or is there a support for transferring ownership to the team (or it's buildbot or similar) later on?

I couldn't find an answer for this. I am not too familiar with `gpg` or how it's used in Sandstorm marketplace so I couldn't answer this myself.

> What would be the best way to integrate sandstorm marketplace app with tiddlywiki development?

I'm most likely thinking too far ahead, but seeing what has happened to for example Ghost on Sandstorm marketplace (last I checked, it had fallen way behind in update schedule).

Even with the above issues open, I'd say it's better to publish something to the marketplace and work on any issues that come up later. I'll try to test this repo of yours later today and I'll comment back on this issue with the results.

**Comment by jeau** [comment]
Hi Joonas

I have benefited from your patch without knowing ;) In fact, I was surprised that it works (apparently almost without too many bugs) as easily.

I agree with you, we need to maintain the application synced with the development of TW5. I don't feel very qualified to do that. My technical level is quite limited, the quality of my english is poor and I'm not familiar with `gpg`. However we could join our efforts.

Should be discussed with @Jermolene. The solution may lie in the proposal of Tony late last year on the Google Group TiddlyWiki:

> After TiddlyWiki is packaged as an app, I think Jeremy or another TW representative developer can sign off as the upstream author as Sandstorm is open to adding additional licenses like BSD3

https://groups.google.com/d/msg/tiddlywiki/JArRJpuYWkI/SUkqBM2VBAAJ

I also agree with you, it's better to attempt publish (after tests) our app to the marketplace and work together on any issues that come up later.

regards
Thierry

**Comment by koivunej** [comment]
Just tested the package yielding one PR and a possible issue with chromium and the svg icons.

Tested:
- create, update tiddler
- create, delete tiddler

No problems, all HTTP traffic seem'd to work nicely. The package is currently at least as functional as mine was once I got the patch submitted.

**Comment by koivunej** [comment]
Like [in comment](https://github.com/jeau/tiddlywiki-sandstorm/issues/3#issuecomment-243061959) for issue #2 I noted:

> Sharing the grain worked, we can scratch that off as tested as well. Again I created two tiddlers, removed the first and updated the other.

To recap the testing:
- [x] create tiddler
- [x] update tiddler
- [x] delete tiddler
- [x] sharing a grain via invite (create/update/delete)

**Comment by koivunej** [comment]
Regarding your [previous comment and the Google groups discussion](https://github.com/jeau/tiddlywiki-sandstorm/issues/1#issuecomment-242974831) I do not really understand what is the issue with BSD3 or why it was brought up in:

> After TiddlyWiki is packaged as an app, I think Jeremy or another TW representative developer can sign off as the upstream author as Sandstorm is open to adding additional licenses like BSD3

I can't really see it as a blocker either way?

> I agree with you, we need to maintain the application synced with the development of TW5. I don't feel very qualified to do that. My technical level is quite limited, the quality of my english is poor and I'm not familiar with gpg.

Well the same goes for me with gpg. Of course to start out the syncing could be manual.

> However we could join our efforts.

I'd like to help out as much as I can. Getting and keeping tiddlywiki in sandstorm appstore is something I have wanted for some time now :)

I'd like to suggest you to [revive that thread you found on google groups](https://groups.google.com/forum/#!msg/tiddlywiki/JArRJpuYWkI/SUkqBM2VBAAJ) and present this work of yours with the problems we see (`gpg` for one at least). Perhaps there's someone with `gpg` confidence already in the team/groups who could easily take on the task of publishing or guiding us? If not, I think I might be able to understand the publishing process but wouldn't keep that as a first option :)

**Comment by jeau** [comment]
I have copied the all sentence, I don't think the choice BSD3 is imperative. There are no reasons. Before read this proposal, I have chosen Apache3 which seems is the Sandstorm dev's choice. Ok, I will send a message on TW5 Group.
